lib/services/android-project-service.ts

```diff
@@ -257,13 +258,17 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			buildOptions.unshift("buildapk");
 
-			this.$childProcess.on(constants.BUILD_OUTPUT_EVENT_NAME, (data: any) => {
+			const handler = (data: any) => {
 				this.emit(constants.BUILD_OUTPUT_EVENT_NAME, data);
-			});
+			};
 
-			await this.executeGradleCommand(this.getPlatformData(projectData).projectRoot, buildOptions,
-				{ stdio: buildConfig.buildOutputStdio || "inherit" },
-				{ emitOptions: { eventName: constants.BUILD_OUTPUT_EVENT_NAME }, throwError: true });
+			await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
+				this.$childProcess,
+				handler,
+				this.executeGradleCommand(this.getPlatformData(projectData).projectRoot,
```

Comment:
In case this method fails before returning the Promise (for example in case the implementation is:
```TypeScript
myMethod(): Promise<any> {
    // some sync operation that throws error
    return new Promise<any>((resolve, reject) => {
        // execute some operations
        resolve();
    });
}
```
current code will fail during evaluation of method arguments and will never emit error event for the operation. Is this expected?

Note: I know we are using `async` keyword in such cases that will always generate Promise for us and we should never end up in a case where we fail before creating the Promise, but anyway, I want to be sure that's the behaviour we want.